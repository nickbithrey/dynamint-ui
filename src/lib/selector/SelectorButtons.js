import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Button from '../Button';
import ButtonsContainer from '../ButtonsContainer';

const SelectorButtons = ({buttons, selection}) => (
	<ButtonsContainer>
		{buttons.map(btn => btn.build()).filter(btn => btn.condition(selection)).map((btn, i) => renderButton(btn, i, selection))}
	</ButtonsContainer>
);

function renderButton(btn, i, selection) {
	let to = {
		pathname: btn.pathname,
		state: btn.stateFn(selection)
	};
	return <Link key={i} to={to}><Button text={btn.text} /></Link>
}

SelectorButtons.defaultProps = {
	buttons: []
};

SelectorButtons.propTypes = {
	buttons: PropTypes.array.isRequired,
	selection: PropTypes.object.isRequired
};

export default SelectorButtons;