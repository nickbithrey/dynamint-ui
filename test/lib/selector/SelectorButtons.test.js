import React from 'react';
import { shallow } from 'enzyme';

import SelectorButtons from '~/lib/selector/SelectorButtons';

describe('<SelectorButtons /> rendering', () => {

	const props = {
		buttons: [
			{
				build: () => ({
					pathname: 'path',
					stateFn: sel => ({}),
					text: 'btnText',
					condition: sel => true,
				})
			}
		],
		selection: {
			count: 0
		}
	};
	
	let wrapper;
	
	beforeEach(() => {
		wrapper = shallow( <SelectorButtons {...props} /> );
	});

	it('can render standard component with records', () => {
		expect(wrapper).toMatchSnapshot();
	});
		
});