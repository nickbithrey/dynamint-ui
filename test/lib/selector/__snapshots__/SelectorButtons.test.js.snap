// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectorButtons /> rendering can render standard component with records 1`] = `
<ButtonsContainer>
  <Link
    key="0"
    replace={false}
    to={
      Object {
        "pathname": "path",
        "state": Object {},
      }
    }
  >
    <Button
      text="btnText"
      type="standard"
    />
  </Link>
</ButtonsContainer>
`;
